highfidelity/hifi
# Add a visualization of cacheable resource usage

- Move plotperf qml visualization to a shared folder
- Changes plotperf to poll and redraw at 10fps
- Clean ResourceCache to avoid exposing dangerous methods
- Add counter properties to ResourceCache for count and size, total and cached
- Add resources property to ResourceCache to list active resource URLs
- Expose Animation, Model, Texture, Sound caches to QML/JS
- Add cacheStats.js to show cache count/size

---

#### Intent
We are seeing slow rendering because of high resource usage. This is meant to show if resources are being freed or not, whether they are being held by something or cached appropriately, and whether the caches are releasing resources appropriately
(See: https://app.asana.com/0/74051501495175/105559909158465.)

#### Testing

- Ensure that renderStats.js still works.
- Ensure that cacheStats.js works as expected. It should show resource usage. Take it for a spin across a few domains :smile:.
- Check TextureCache.resources in the JS console (or ModelCache, AnimationCache, SoundCache) and see the loaded/loading resources in your domain.

*NOTE: You may notice a large amount of texture usage (400Mb) without it being cached after visiting the playa. This is fixed in #7546, and is exactly the kind of thing this tool is meant to help find.*
